featuretools/entityset/timedelta.py

```diff
@@ -44,48 +44,51 @@ class Timedelta(object):
         "o": "Observations",
         "w": "Weeks",
         "Y": "Years",
-        'mo': "Months"
-    }
-
-    _convert_to_days = {
-        "w": 7
+        "mo": "Months"
     }
 
     _readable_to_unit = {v.lower(): k for k, v in _readable_units.items()}
 
-    def __init__(self, value, unit=None):
+    def __init__(self, value, unit=None, delta_obj=None):
         """
         Args:
-            value (float, str) : Value of timedelta, or string providing
-                both unit and value.
+            value (float, str, dict) : Value of timedelta, string providing
+                both unit and value, or a dictionary of units and times.
             unit (str) : Unit of time delta.
         """
+
         # TODO: check if value is int or float
         if is_string(value):
             from featuretools.utils.wrangle import _check_timedelta
             td = _check_timedelta(value)
-            value, unit = td.value, td.unit
-
-        self.value = value
-        self._original_unit = None  # to alert get_name that although we converted the unit to 'd' it was initially
-        unit = self._check_unit_plural(unit)
-        assert unit in self._readable_units or unit in self._readable_to_unit
-        if unit in self._readable_to_unit:
-            unit = self._readable_to_unit[unit]
-
-        # weeks
-        if unit in self._convert_to_days:
-            self._original_unit = unit
-            self.value = self.value * self._convert_to_days[unit]
-            unit = 'd'
-
-        self.unit = unit
-        self.delta_obj = self.get_unit_type()
+            self.times = td.times
```

Comment:
Can we add helper functions to check value and fixed_units? This would make init more clear.